PowerballApi.Tests/PowerballRepositoryTests.cs

```diff
@@ -1,140 +1,271 @@
+            });
+
+            try
+            {
+                _sut.Get(date);
+            }
+            catch (Exception ex)
+            {
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_InvalidAfterDate_ThrowException()
+        {
+            const string dateAfter = "12/50/2011";
+            const string dateBefore = "12/20/2013";
+
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(dateBefore),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+
+            {
+                drawing
+            });
+
+            try
+            {
+                _sut.GetByRange(dateAfter, dateBefore);
+            }
+            catch (Exception ex)
+            {
+
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_InvalidBeforeDate_ThrowsException()
+        {
+            const string dateAfter = "01/15/2015";
+            const string dateBefore = "01/44/2016";
+
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(dateAfter),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+
+            {
+                drawing
+            });
+
+            try
+            {
+                _sut.GetByRange(dateAfter, dateBefore);
+            }
+            catch (Exception ex)
+            {
+
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_ValidDates_ReturnsDrawingsInRange()
+        {
+            const string dateAfter = "01/01/2017";
+            const string dateBefore = "01/31/2017";
+
+            var drawingList = new List<PowerballSet>
+            {
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/01/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/10/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/21/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(drawingList);
+
+            var result = _sut.GetByRange(dateAfter, dateBefore);
+
```

Comment:
You may want to also assert the result isn't null. This way, if null is returned, you will get a test failure rather than a runtime exception.